fdbclient/RESTUtils.actor.cpp

```diff
@@ -207,7 +235,14 @@ void RESTUrl::parseUrl(const std::string& fullUrl) {
 		uint8_t foundSeparator = 0;
 		StringRef hostPort = t.eatAny("/?", &foundSeparator);
 		if (foundSeparator == '/') {
-			this->resource = t.eat("?").toString();
+			StringRef resourceRef = t.eat("?");
+			if (!resourceRef.empty()) {
```

Comment:
`eat()` consumes but does not output the found `separator`, which means when you find a `/` you need to seed your resource (aka path in rfc2396) with `/` and then add the rest.

Instead of the changes here, I suggest this, which fixes this issue and the one described below.

```
		StringRef hostPort = t.eatAny("/?", &foundSeparator);
+		this->resource = "/";
		if (foundSeparator == '/') {
-			this->resource = t.eat("?").toString();
+			this->resource += t.eat("?").toString();
```

The other issue is that for URLs with an empty `path` such as
```
http://www.foo.com
http://www.foo.com?
http://www.foo.com?foo=bar
```
it is common for REST/HTTP clients, such as `curl`, to use `/` as the path so that the outbound request is valid, so that would be a good behavior here too and the above does this.